// INCLUDE. --------------------------------------------------------------------
//------------------------------------------------------------------------------
#include <functional>


#include "ver.h"

#include "drw.h"
#include "std.h"
#include "wgt.h"

#include "RAbout.h"
#include "ui_RAbout.h"


//Глобальные переменные.--------------------------------------------------------
//------------------------------------------------------------------------------

/* ДОПОЛНИТЕЛЬНЫЕ ФУНКЦИИ. ****************************************************/
/******************************************************************************/

//Показать форму.---------------------------------------------------------------
//------------------------------------------------------------------------------
int RAbout::view(void) { return this->exec(); }

//Показать форму, с изменением баннера.-----------------------------------------
//------------------------------------------------------------------------------
int RAbout::view(QString styleSheet)
    { style = styleSheet; return this->view(); }

/* RAbout *********************************************************************/
/******************************************************************************/

//Начальная инициализация.------------------------------------------------------
//------------------------------------------------------------------------------
void RAbout::startInit(void){
    start = QDateTime::currentDateTime();
    ui->setupUi(this); TO_CENTER(DESKTOP);
    this->setWindowFlags(Qt::WindowCloseButtonHint);
}// StartInit

//Выполняется при старте.-------------------------------------------------------
//------------------------------------------------------------------------------
RAbout::RAbout(QString styleSheet, QWidget *parent)
: QDialog(parent), style(styleSheet), ui(new Ui::RAbout)
{
    startInit();
}// RAbout

//Выполняется при старте.-------------------------------------------------------
//------------------------------------------------------------------------------
RAbout::RAbout(QWidget *parent) : QDialog(parent), ui(new Ui::RAbout){
    startInit();
}// RAbout

//Выполняется при показе формы.-------------------------------------------------
//------------------------------------------------------------------------------
void RAbout::showEvent(QShowEvent*){
    if(style != ui->frImg->styleSheet()) ui->frImg->setStyleSheet(style);

#ifdef QT_DEBUG
    #define CLR Qt::red
    #define ADD tr(" отладка.")
#else
    #define CLR Qt::green
    #define ADD tr(" релиз.")
#endif

    // Выполняется однократно.
    if(FIRST_RUN) {
        ui->lbNme->setText(
            PRODUCTNAME_STR
          + tr(" / пуск: ") + start.toString("yyyy.MM.dd hh:mm") );
        ui->lbVer->setText(VERSION_STR + ADD);
        ui->lbUpd->setText(LAST_UPDATE_STR);

        BACKGROUND_CLR(ui->lbVer, CLR);
    }

#undef CLR
#undef ADD
}// showEvent

//Выполняется по завершении.-------------------------------------------------
//---------------------------------------------------------------------------
RAbout::~RAbout(){
    delete ui;
}// ~RAbout

//---------------------------------------------------------------------------
